test_conformance/device_execution/enqueue_flags.cpp

```diff
@@ -676,7 +672,7 @@ int test_enqueue_flags(cl_device_id device, cl_context context, cl_command_queue
     cl_uint i;
     cl_int err_ret, res = 0;
     clCommandQueueWrapper dev_queue;
-    cl_int kernel_results[MAX_GWS] = { -1 };
+    cl_int kernel_results[MAX_GWS] = { 0 };
```

Comment:
Why did you change that line? My understanding of the validation logic below is that both 0 and 1 are expected values in the output. Isn't it better to initialise the results array with a value that can never be valid? I'm afraid this might mask failures.

Reply:
The value is being overwritten in kernel anyway. But the way some kernels re-initialize res array causes data races. So, after some discussions here and there I decided to change this value to 0. This should not seriously affect the test. But definitely fixes data races.

Reply:
Not sure I'm following. Don't we want to guarantee that the entirety of the array is filled by the kernels? It seems to me that we may be hiding issues in the kernels with this change.

>This should not seriously affect the test. But definitely fixes data races.

I can't reconcile these two statements :).

Reply:
The original issue #40 contains detailed explanations with pictures https://github.com/KhronosGroup/OpenCL-CTS/issues/40#issuecomment-456696470 

This is the approach we agreed on with @b-sumner offline. I'll add you to the mail thread.